### ISSUE TITLE
Fail when some targets do not have provider 'cc'

### ISSUE THREAD

**Issue by mapx** [grailbio/bazel-compilation-database#8]
Take the example referred in https://blog.bazel.build/2017/02/27/protocol-buffers.html

To make the error recur on top of https://github.com/cgrushko/proto_library , one may add a main file `src/main.cc`:
```cpp
#include "src/person.pb.h"

int main() { demo::Person person; }
```

And append the following lines to `src/BUILD`:
```python
cc_binary(
    name = "main",
    srcs = [
        "main.cc",
    ],
    deps = [
        ":person_cc_proto",
    ],
)
```

Then `generate.sh` fails:

```sh
ERROR: /tmp/a_project/BUILD:44:1: in //:bazel-compilation-database-0.1/aspects.bzl%compilation_database_aspect aspect on proto_library rule //src:zip_code_proto:
Traceback (most recent call last):
        File "/tmp/a_project/src/BUILD", line 44
                //:bazel-compilation-database-0.1/aspects.bzl%compilation_database_aspect(...)
        File "/tmp/a_project/bazel-compilation-database-0.1/aspects.bzl", line 52, in _compilation_database_aspect_impl
                target.cc
<target //src:zip_code_proto> (rule 'proto_library') doesn't have provider 'cc'.
ERROR: Analysis of aspect '/bazel-compilation-database-0.1/aspects%compilation_database_aspect of //src:main' failed; build aborted.
```

When it hits:
```python
proto_library(  # line 44
    name = "zip_code_proto",
    srcs = ["zip_code.proto"],
)
```
BTW, if one run
```sh
./bazel-compilation-database-0.1/generate.sh
```
rather than 
```sh
bazel-compilation-database-0.1/generate.sh
```
It fails with:
```sh
ERROR: invalid target name './bazel-compilation-database-0.1/aspects.bzl': target names may not contain '.' as a path segment.
ERROR: Analysis of aspect '/./bazel-compilation-database-0.1/aspects%compilation_database_aspect of //src:main' failed; build aborted.
```

Thanks :)

**Comment by siddharthab** [comment]
Ah, it does not run with the cc_proto_library dependencies yet. Let me see if I can support that.

**Comment by siddharthab** [comment]
Should be fixed now. I tested with your example.

Let me know if it does not work.

**Comment by siddharthab** [comment]
Fix was in 8869755d2d558cc1c5a3e9b265e58431db0bbcf9.
